src/main/java/org/swisspush/redisques/util/QueueConfiguration.java

```diff
@@ -2,34 +2,64 @@
 
 import io.vertx.core.json.JsonObject;
 
-import java.util.List;
+import java.util.Arrays;
+import java.util.regex.Pattern;
 
 public class QueueConfiguration {
-    
-    static final String PROP_PATTERN = "pattern";
-    static final String PROP_RETRY_INTERVALS = "retryIntervals";
-    
-    private String pattern;
-    
-    private List<Integer> retryIntervals;
 
-    @SuppressWarnings("unused")
-    public QueueConfiguration() {
-        // required to be deserialized by io.vertx.core.json.JsonObject.mapTo(type)
-    }
+    /**
+     * This configuration applies to queues with names matching this RegEx
+     */
+    private Pattern pattern;
+
+    /**
+     * after a failed de-queuing the next try is delayed by the first entry (values are seconds)
+     * The 2nd failed de-queue takes the seconds array-entry for delay, the 3rd takes the 3rd, etc...
+     */
+    private int[] retryIntervals;
+
+    /**
+     * EN-queuing speed can be throttled by delaying the "ok" response.
+     * This is a simple linear factor. E.g. when set to "0.3" this means that
+     * enqueuing is delayed for 0.3 seconds when queue is of length 1000 (0.6 seconds when length is 2000 etc...)
+     * "0" means: turn of this feature
+     */
+    private float enqueueDelayMillisPerSize = 0f;
```

Comment:
I am a bit confused about the naming. The contains `DelayMillis` but the values are float and documented as seconds. Is the `PerSize` factor always 1000 or can this be configured?

Reply:
"size" has no 'unit'. The delay _has_ a unit, namely _seconds_ (or here: _ms_ aka _milliseconds_ or _millis_)
So the factor has the unit _millis_

Reply:
Your proposal of a "minimum delay" sounds good.
Though there are no requirements at the moment. Feel free to create an Issue ;-)

Reply:
ok, what about calling the property just `enqueueDelayFactor`? The property itself does not provide the delay in milliseconds. Only the mulitplication of the factor with the current queue size does.

Reply:
Unit of this factor **is** milliseconds - i really prefer to have the unit in the name.
Otherwise everyone is enforced to read JavaDoc. We all hate "x-timeout: 10" and nobody knows if we talk about "10 seconds" or "10 millis"

Reply:
Renamed to `enqueueDelayFactorMillis` and improved JavaDoc of this property.
Are you happy with this?

Reply:
ok, another proposal: `enqueueDelayMillisFactor`. I would remove the "PerSize" part because this is covered by the factor.